garden-backend-service/README.md

```diff
@@ -35,64 +35,59 @@ garden-backend-service/
     DB_USERNAME="garden_dev"
     DB_PASSWORD="your_password"
-    DB_ENDPOINT="host.docker.internal"
-    
-where the AWS access key variables correspond to the `garden_lightsail_user_dev` IAM user (which has permission to read the AWS secret). If you provide any additional variables which are also present in the `garden-backend-env-vars/dev` secret, the one you set in the .env file will take priority. 
-
-### Local PostgreSQL setup 
-You'll need postgres installed locally and configured to allow connections from the docker container. 
-
-0. install/start postgres server
-  - on mac, [Postgres.app](https://postgresapp.com/downloads.html) is probably the easiest option. Make sure the CLI tools installed with the app are on your path. 
-1. create a `garden_db_dev` database and a privileged user with the postgres CLI:
-
-``` sql
-$ psql postgres
-psql (16.2 (Homebrew))
-Type "help" for help.
-
-postgres=# CREATE DATABASE garden_db_dev;
-CREATE DATABASE
-postgres=# CREATE USER garden_dev WITH ENCRYPTED PASSWORD 'your_password';
-CREATE ROLE
-postgres=# GRANT ALL PRIVILEGES ON DATABASE garden_db_dev TO garden_dev;
-GRANT
-```
+    DB_ENDPOINT="dev-db" # hostname of the db container in compose.yaml
+    POSTGRES_DB="garden_db_dev"
+    POSTGRES_PASSWORD=$DB_PASSWORD
+
+where the AWS access key variables correspond to the `garden_lightsail_user_dev` IAM user (which has permission to read the AWS secret). If you provide any additional variables which are also present in the `garden-backend-env-vars/dev` secret, the one you set in the .env file will take priority.
+
+POSTGRES variables are used by Docker to configure the database container.
 
-2. edit your `postgresql.conf` to set `listen_addresses`:
-  - the location of this file depends on how you installed postgres; you can find it in Postgress.app via "server settings" or at `/opt/homebrew/var/postgresql@16/postgresql.conf` if installed with homebrew.
-  - uncomment the line `listen_addresses = 'localhost'` and set it to `'*'` to allow connections from any IP. 
-    - Alternatively, you can set it to `'localhost,172.17.0.1'` to allow connections only from localhost and docker containers (where 172.17.0.1 is the docker subnet gateway -- run `docker network inspect bridge`to double check).
-3. edit `pg_hba.conf` (same dir as `postgresql.conf`) to add a connection rule for docker containers:
-  - paste the following line at the bottom (assuming default docker subnet 172.17.0.0/16):
+#### Docker
+Run the API and database containers locally  using `docker compose`:
 
+``` sh
+docker compose up
+
+# or run in the background
+docker compose up -d
 ```
-host    all             all             172.17.0.0/16           md5
+
+Visit http://localhost:5500/docs and behold!
+
+Docker compose maps ./src, ./tests, and ./.env into the container so you can edit files locally
+and see the changes immediately reflected in the running container.
+
+Tear down and cleanup the containers:
+``` sh
+docker compose down
```

Comment:
is there an easy way to have docker compose automatically clean up the containers (like regular `docker run --rm`) when the app container exits / on a Ctrl-C? if it's not easy I wouldn't worry about it

Reply:
also feel free to edit/delete the `run-dev-server.sh` script now that we're replacing it with docker compose